openpype/pipeline/version_start.py

```diff
@@ -0,0 +1,32 @@
+from openpype.lib.profiles_filtering import filter_profiles
+
+
+def get_versioning_start(
+    project_settings=None,
+    host=None,
+    task_name=None,
+    task_type=None,
+    family=None,
+    subset=None
+):
```

Comment:
This is how it should be. You should expect the required arguments without any default value. Project name in this case is MUST HAVE. Without project name there is no reason to care about versioning. In that case project settings can be received without having them available. Host name is must have, because that's the minimum for this feature, and you can't think about versions without a host...

```suggestion
def get_versioning_start(
    project_name,
    host_name,
    task_name=None,
    task_type=None,
    family=None,
    subset=None,
    project_settings=None,
):
```

Reply:
Default function implementations should never use current context. If they get project settings `None` they should do nothing. From API stand of point it's unclear "what actually the function needs to work properly". Minimum requirement is project name and host name. Project settings are optional to make the function faster.

Reply:
Also use `host_name` where it is host name please.